### ISSUE TITLE
QMK-Like Toggle Layer

### ISSUE THREAD

**Issue by hadeedji** [kmonad/kmonad#520]
I'm trying to make a button that behaves like `TG(Layer)` does in QMK. It adds the layer to the stack if it wasn't already there, and removes it if it was. A combination of `layer-add` and `layer-rem` in KMonad essentially, with some logic.

Is there a way to achieve this? If not maybe this can be a feature request.

**Comment by MuhammedZakir** [comment]
What I do is to map a key in base layer to `layer-add <layer>` and map the same key **inside** the `<layer>` to `layer-rem <layer>`.

**Comment by hadeedji** [comment]
Definitely a solution. Just that the way I used to do this in QMK, and how I want to do it here is that the layer toggle key isn't on the base layer or `<layer>`, but on a third layer. 
One way I can see how to do it is duplicate this third layer for `<layer>`. But I wouldn't like that and it would get crazy if I had to add a few more toggles.

**Comment by MuhammedZakir** [comment]
Why can't you do this:
- In "third layer", map keyX to `layer-add <layer>`.
- In `<layer>`, map keyX to `layer-rem <layer>`.

Only difference compared to `TG(layer)` is that you have to define 2 buttons.

P.S. I am not against adding `TG(layer)`-like button.

**Comment by hadeedji** [comment]
That also does not work for me, because it's not feasible to map keyX to something in `layer`. I want keyX to be a very accessible key, maybe on the homerow, and the third layer to be a layer that's rarely used so the buttons on homerow can be used however.

I made the issue to confirm that something like this doesn't exist and I wasn't missing something. Seems like it doesn't. If enough people want it maybe it can be implemented someday. Thanks for your help.

**Comment by hadeedji** [comment]
I'll close the issue. I did not find a neat way to do what I wanted, but learned to live without it.

**Comment by MuhammedZakir** [comment]
It's better to keep this open, so we can implement this. No eta though. :-)

**Comment by hadeedji** [comment]
Right. It's my first issue so I didn't know how these are handled generally. Let's leave it open then.
